samples/language/FailStatement.qs

```diff
@@ -0,0 +1,24 @@
+/// # Sample
+/// Fail Statement
+///
+/// # Description
+/// A fail statement collects information about the current state of the program,
+/// and then terminates execution entirely. The presented information will be
+/// presented to the user along with the message specified with the fail statement.
+namespace MyQuantumApp {
+    @EntryPoint()
+    operation Main() : Unit {
+        if 0 / 0 == 0 {
+            // Fail statements are useful when representing control flow
+            // for unreachable lines
+            fail "Division by zero is impossible, how did we get here?"
+        }
```

Comment:
```suggestion
        use q = Qubit();
        X(q);
        if M(q) == Zero {
            // Fail statements are useful when representing control flow
            // for unreachable lines.
            fail "Measurement should have been `One`, how did we get here?"
        }
        Reset(q);
```

Reply:
Samples should succeed when running.

The suggestion still illustrates the point without causing a runtime error.